packages/app-project/stores/Project.js

```diff
@@ -90,5 +92,23 @@ const Project = types
       return self.links['active_workflows'].includes(workflowId)
     }
   }))
+  .preProcessSnapshot(snapshot => {
+    let workflowFromCookie = ''
+    if (typeof document !== 'undefined') {
+      workflowFromCookie = getCookie('workflow_id')
+    }
+    return {
+      ...snapshot,
+      selectedWorkflow: workflowFromCookie
+    }
+  })
+  .actions(self => ({
+    setSelectedWorkflow(workflowId) {
+      if (typeof document !== 'undefined') {
+        self.selectedWorkflow = workflowId
+        document.cookie = `workflow_id=${workflowId}; path=/projects/${self.slug}; domain=zooniverse.org; SameSite=Strict`
```

Comment:
`SameSite=Strict` means the cookie is only read while navigating the `zooniverse.org` domain. For example, it will work when going to Talk and back but it will be ignored if I enter the site from a link in an email, or a link on an external site. In those latter cases, I’ll be asked to choose a workflow, unless one is specified in the link.